core/Tracker/VisitorRecognizer.php

```diff
@@ -87,7 +88,11 @@ public function findKnownVisitor($configId, VisitProperties $visitProperties, Re
 
         $isVisitorIdToLookup = !empty($idVisitor);
 
-        if ($isVisitorIdToLookup) {
+        if (PrivacyManager::shouldAnonymiseFingerprint()) {
+            $isVisitorIdToLookup = false;
+            $idVisitor = false;
```

Comment:
here we wouldn't want to change the logic in "findKnownVisitor" when cookies are used and "limit fingerprint" feature is activated, then the cookie visitor id would need to be used to match visitors accurately. (we assume eg. consent was given when cookies are received by tracking api). 

(Otherwise this would break legit use cases and changes/bc break tracking API, especially as the setting is global and affects all sites)